drivers/interrupt_controller/intc_exti_stm32.c

```diff
@@ -189,217 +116,364 @@ static int stm32_exti_enable_registers(void)
-	__ASSERT_NO_MSG(irqnum != 0xFF);
+/**
+ * @brief Disables rising trigger for specified EXTI line
+ *
+ * @param line EXTI line
+ * @returns 0 on success, -EINVAL if @p line is invalid
+ */
+static int stm32_exti_disable_rising_trig(uint32_t line)
+{
+	int ret = 0;
+	const uint32_t line_num = exti_linenum_to_ll_exti_line(line);
+
+	if (line_num < 32U) {
+		EXTI_DISABLE_RISING_TRIG_0_31(line);
+#ifdef CONFIG_EXTI_STM32_HAS_64_LINES
+	} else if (line_num < 64U) {
+		EXTI_DISABLE_RISING_TRIG_32_63(line);
+#endif /* CONFIG_EXTI_STM32_HAS_64_LINES */
+#ifdef CONFIG_EXTI_STM32_HAS_96_LINES
+	} else if (line_num < 96U) {
+		EXTI_DISABLE_RISING_TRIG_64_95(line);
+#endif /* CONFIG_EXTI_STM32_HAS_96_LINES */
+	} else {
+		LOG_ERR("Invalid line: %d", line);
+		ret = -EINVAL;
+	}
 
-	/* Enable requested line interrupt */
-#if defined(CONFIG_SOC_SERIES_STM32H7X) && defined(CONFIG_CPU_CORTEX_M4)
-	LL_C2_EXTI_EnableIT_0_31(line);
-#else
-	LL_EXTI_EnableIT_0_31(line);
-#endif
+	return ret;
+}
+
+/**
+ * @brief Enables falling trigger for specified EXTI line
+ *
+ * @param line EXTI line
+ * @returns 0 on success, -EINVAL if @p line is invalid
+ */
+static int stm32_exti_enable_falling_trig(uint32_t line)
+{
+	int ret = 0;
+	const uint32_t line_num = exti_linenum_to_ll_exti_line(line);
+
+	if (line_num < 32U) {
+		EXTI_ENABLE_FALLING_TRIG_0_31(line);
+#ifdef CONFIG_EXTI_STM32_HAS_64_LINES
+	} else if (line_num < 64U) {
+		EXTI_ENABLE_FALLING_TRIG_32_63(line);
+#endif /* CONFIG_EXTI_STM32_HAS_64_LINES */
+#ifdef CONFIG_EXTI_STM32_HAS_96_LINES
+	} else if (line_num < 96U) {
+		EXTI_ENABLE_FALLING_TRIG_64_95(line);
+#endif /* CONFIG_EXTI_STM32_HAS_96_LINES */
+	} else {
+		LOG_ERR("Invalid line: %d", line);
+		ret = -EINVAL;
+	}
 
-	/* Enable exti irq interrupt */
-	irq_enable(irqnum);
+	return ret;
 }
 
-void stm32_gpio_intc_disable_line(stm32_gpio_irq_line_t line)
+/**
+ * @brief Disables falling trigger for specified EXTI line
+ *
+ * @param line EXTI line
+ * @returns 0 on success, -EINVAL if @p line is invalid
+ */
+static int stm32_exti_disable_falling_trig(uint32_t line)
 {
-#if defined(CONFIG_SOC_SERIES_STM32H7X) && defined(CONFIG_CPU_CORTEX_M4)
-	LL_C2_EXTI_DisableIT_0_31(line);
-#else
-	LL_EXTI_DisableIT_0_31(line);
-#endif
+	int ret = 0;
+	const uint32_t line_num = exti_linenum_to_ll_exti_line(line);
+
+	if (line_num < 32U) {
+		EXTI_DISABLE_FALLING_TRIG_0_31(line);
+#ifdef CONFIG_EXTI_STM32_HAS_64_LINES
+	} else if (line_num < 64U) {
+		EXTI_DISABLE_FALLING_TRIG_32_63(line);
+#endif /* CONFIG_EXTI_STM32_HAS_64_LINES */
+#ifdef CONFIG_EXTI_STM32_HAS_96_LINES
+	} else if (line_num < 96U) {
+		EXTI_DISABLE_FALLING_TRIG_64_95(line);
+#endif /* CONFIG_EXTI_STM32_HAS_96_LINES */
+	} else {
+		LOG_ERR("Invalid line: %d", line);
+		ret = -EINVAL;
+	}
+
+	return ret;
 }
 
-void stm32_gpio_intc_select_line_trigger(stm32_gpio_irq_line_t line, uint32_t trg)
+static inline int stm32_exti_intc_select_line_trigger(uint32_t line, uint32_t trg)
```

Comment:
Nit: name can be shortened
```suggestion
static inline int stm32_exti_select_line_trigger(uint32_t line, uint32_t trg)
```